pkg/proxy/proxy.go

```diff
@@ -137,7 +137,12 @@ func (s *SQLServer) onConn(ctx context.Context, conn net.Conn) {
 		delete(s.mu.clients, connID)
 		s.mu.Unlock()
 
-		logger.Info("connection closed", zap.Error(clientConn.Close()))
+		if err := clientConn.Close(); err != nil {
+			// Close() always reports `tls: failed to send closeNotify alert (but connection was closed anyway)`
+			// when the client quits unexpectedly.
+			logger.Debug("close connection fails", zap.Error(err))
+		}
+		logger.Info("connection closed")
```

Comment:
Should be able to do something like:

```
@@ -137,10 +138,9 @@ func (s *SQLServer) onConn(ctx context.Context, conn net.Conn) {
                delete(s.mu.clients, connID)
                s.mu.Unlock()

-               if err := clientConn.Close(); err != nil {
-                       // Close() always reports `tls: failed to send closeNotify alert (but connection was closed anyway)`
-                       // when the client quits unexpectedly.
-                       logger.Debug("close connection fails", zap.Error(err))
+               // ignore 'broken pipe' error, it may be caused by the server
+               if err := clientConn.Close(); err != nil && !errors.Is(err, syscall.EPIPE) {
+                       logger.Error("close connection fails", zap.Error(err))
                }
                logger.Info("connection closed")
                metrics.ConnGauge.Dec()
```

Reply:
BTW, you should not need to log `connection closed` if there `fail to close connection` is logged as `Error`.